cpp/src/io/parquet/chunk_dict.cu

```diff
@@ -151,11 +150,13 @@ __global__ void __launch_bounds__(block_size)
           case Type::INT96: return 12;
           case Type::FLOAT: return 4;
           case Type::DOUBLE: return 8;
-          case Type::BYTE_ARRAY:
-            if (data_col.type().id() == type_id::STRING) {
+          case Type::BYTE_ARRAY: {
+            auto const col_type = data_col.type().id();
+            if (col_type == type_id::STRING) {
               // Strings are stored as 4 byte length + string bytes
               return 4 + data_col.element<string_view>(val_idx).size_bytes();
             }
+          }
```

Comment:
Should there be a `break` here?

Reply:
I looked at this and I wasn't sure. I left the original behavior, but this could be a latent bug. I can run tests with a break and see. I assume it would need to be a return or an assert that there aren't other types coming through here. One of the things I intended to ask Devavret...

Reply:
I think it's meant to fall through and hit the default case. Maybe the right thing to do would be to CUDF_EXPECT a string type here and dec128 type below.

Reply:
Hmm. Not `break`, right? 
If `col_type != type_id::STRING`, shouldn't we be throwing `CUDF_UNREACHABLE`?

Reply:
(Argh. Sorry for the out-of-order comments. Looks like I'm missing updates in the Github comments.) 
It appears that this problem was preexisting, and not introduced in this change.

Reply:
device code doesn't support `CUDF_EXPECTS`. :(

Reply:
> It appears that this problem was preexisting, and not introduced in this change.

yes, but we should either clean it up or file an issue.

Reply:
I vote followup.  It takes a long time for this code to get thoroughly exercised so there's no guarantee we'd find any issues with the time left.

Reply:
lol, didn't even pay attention to whether it's a device code :face_with_spiral_eyes: 
Either way, not blocking the PR on this, it's unrelated.

Reply:
> I vote followup. It takes a long time for this code to get thoroughly exercised so there's no guarantee we'd find any issues with the time left.

That's a good point. Should I pull the expects to prevent any surprises until we can vet the change better?

Reply:
Yeah I'd say so.

Reply:
I think the new code is correct. Fall through does not make sense here. IMO there's no way dec128 has the same size as FIXED_LEN_BYTE_ARRAY and BYTE_ARRAY.

Reply:
In other words: do whatever is easier, I don't think this is a footgun.

Reply:
Added an issue to ensure we are doing the correct thing here. I agree with you the more I think about it. It would fall through before from `BYTE_ARRAY` to `FIXED_LEN_BYTE_ARRAY` and not match that if and fall through to the exception. This is the same behavior, just more explicit.